mr/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergRecordObjectInspector.java

```diff
@@ -48,7 +49,15 @@ public IcebergRecordObjectInspector(Types.StructType structType, List<ObjectInsp
 
     for (Types.NestedField field : structType.fields()) {
       ObjectInspector oi = objectInspectors.get(position);
-      IcebergRecordStructField structField = new IcebergRecordStructField(field, oi, position);
+
+      IcebergRecordStructField structField;
+      if (caseSensitive) {
+        structField = new IcebergRecordStructField(field, oi, position);
+      } else {
+        Types.NestedField fieldInLowercase = Types.NestedField.of(field.fieldId(), field.isOptional(),
+                field.name().toLowerCase(), field.type(), field.doc());
+        structField = new IcebergRecordStructField(fieldInLowercase, oi, position);
+      }
```

Comment:
nit: After if blocks we tend to leave an extra line

Reply:
Got it!